live-build/misc/ansible-roles/appliance-build.minimal-common/tasks/main.yml

```diff
@@ -233,3 +233,42 @@
   with_items:
     - { key: "net.ipv4.conf.all.promote_secondaries", value: "1" }
     - { key: "net.ipv4.conf.default.promote_secondaries", value: "1" }
+
+#
+# Configure command audit logging
+#
+# We want to record all commands executed on the appliance. Opt out for
+# setsid since all ExecuteUtils.execute wrap each call with setsid.
+#
+# Note we can't use the "service" module, as that fails when running
+# inside the chroot environment, so we enable auditd using "command".
+#
+- apt:
+    name: auditd
+    state: installed
+    update_cache: no
+
+- command: /bin/systemctl enable auditd.service
+
+- lineinfile:
+    dest: /etc/audit/auditd.conf
+    regexp: "{{item.regex}}"
+    line: "{{item.line}}"
+  with_items:
+    - { regex: "^num_logs =", line: "num_logs = 6" }
+    - { regex: "^max_log_file =", line: "max_log_file = 3072" }
+    - { regex: "^max_log_file_action =", line: "max_log_file_action = rotate" }
+    - { regex: "^log_format =", line: "log_format = RAW" }
+
+- lineinfile:
```

Comment:
This seems like it'd be better suited to use the [copy module](https://docs.ansible.com/ansible/latest/modules/copy_module.html), rather than `lineinfile`; e.g.
```
copy:
  dest: /etc/audit/rules.d/audit.rules
  mode: 0644
  content: |
    ## Record all executed commands (excluding setsid)
￼    -a exit,never -S execve -F exe=/usr/bin/setsid
    -a exit,always -S execve

    ## Record command exit failures (execve result is not command result)
    -a exit,always -F a0!=0 -S exit_group
    -a exit,always -F a0!=0 -S exit
```
What do you think?

We already do this sort of thing, to insert whole files, like [here](https://github.com/delphix/appliance-build/blob/master/live-build/misc/ansible-roles/appliance-build.minimal-common/tasks/main.yml#L82).

Reply:
This is appending content to an existing file. Would copy preserve and append here?

Reply:
oh, ok. I didn't realize the file will already exist. in that case, I'd suggest we use [blockinfile](https://docs.ansible.com/ansible/2.3/blockinfile_module.html) since it seems like we're trying to append a whole block, rather than inserting independent lines. This might not have a practical difference, but "blockinfile" feels like a better fit than "lineinfile" to me. What do you think?

Reply:
I confirmed that using copy as above will over-write the original rules file (not append) which is not what we want.

Reply:
Yes. "copy" will replace with the contents provided. "blockinfile" won't do that though; it's similar to "lineinfile" but intended to be used when multiple lines are wanted, like what I think we want to do here.

Reply:
I think blockinfile is what I need (append content to an existing file).
```
+- blockinfile:
+    path: /etc/audit/rules.d/audit.rules
+    insertafter: EOF
+    block: |
+      ## Record all executed commands (excluding setsid)
+      -a exit,never -S execve -F exe=/usr/bin/setsid
+      -a exit,always -S execve
+
+      ## Record command exit failures (execve result is not command result)
+      -a exit,always -F a0!=0 -S exit_group
+      -a exit,always -F a0!=0 -S exit
```